addon/lib/Pioneer.jsm

```diff
@@ -0,0 +1,30 @@
+/* eslint no-unused-vars: ["error", { "varsIgnorePattern": "(EXPORTED_SYMBOLS|Pioneer)" }]*/
+
+const { utils: Cu } = Components;
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(
+  this, "Config", "resource://esper-pioneer-shield-study/Config.jsm"
+);
+XPCOMUtils.defineLazyModuleGetter(
+  this, "PioneerUtils", "resource://esper-pioneer-shield-study/PioneerUtils.jsm"
+);
+
+const Pioneer = {
+  async startup(addonData) {
+    this.utils = new PioneerUtils(Config);
+    const branch = await Pioneer.utils.chooseBranch();
+    this.utils.pioneerAddonMetadata = {
```

Comment:
i would probably just make this:
```js
this.metadata = {
``` 
There's no need to add this to `utils`.

Reply:
Can't use "metadata" in the payload since it conflicts with "metadata" in the parquet schema. 
Adding to utils since this sort of metadata should be handled by utils in the future.

Reply:
I wasn't suggesting adding it to the payload. Just that it should be available at:
`Pioneer.metadata` not `Pioneer.utils.pioneerAddonMetadata`. It seems weird to add it to the utils object.

Reply:
Fixed :)